src/groupeddataframe/splitapplycombine.jl

```diff
@@ -499,25 +512,40 @@ wrap(x::AbstractMatrix) =
     NamedTuple{Tuple(gennames(size(x, 2)))}(Tuple(view(x, :, i) for i in 1:size(x, 2)))
 wrap(x::Any) = (x1=x,)
 
-wrap_table(x::Any) =
+wrap_table(x::Any, ::Val) =
     throw(ArgumentError("return value must not change its kind " *
                         "(single row or variable number of rows) across groups"))
-wrap_table(x::Union{NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}},
-                    AbstractDataFrame,
-                    AbstractVector,
-                    AbstractMatrix}) = wrap(x)
+function wrap_table(x::Union{NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}},
+                             AbstractDataFrame, AbstractMatrix}, ::Val{T}) where T
+    T || throw(ArgumentError("mixing single and multiple column values is not allowed"))
+    wrap(x)
+end
+
+function wrap_table(x::AbstractVector, ::Val{T}) where T
+    T && throw(ArgumentError("mixing single and multiple column values is not allowed"))
+    wrap(x)
+end
+
+function wrap_row(x::Any, ::Val{T}) where T
+    # NamedTuple is not possible in this branch
+    if (x isa DataFrameRow) ⊻ T
+        throw(ArgumentError("mixing single and multiple column values is not allowed"))
+    end
+    wrap(x)
+end
 
-wrap_row(x::Any) = wrap(x)
 # note that also NamedTuple() is correctly captured by this definition
 # as it is more specific than the one below
-wrap_row(x::Union{AbstractVecOrMat, AbstractDataFrame,
-                  NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}}) =
+wrap_row(::Union{AbstractVecOrMat, AbstractDataFrame,
+                 NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}}, ::Val) =
     throw(ArgumentError("return value must not change its kind " *
```

Comment:
Use the same error message as elsewhere?

Reply:
this is what I do. Simply there are two kinds of errors we have to handle:
1. switching from single to multiple rows (or vice versa)
2. switching from one column to multiple columns (or vice versa)

I think the messages are correct in each case. In the case you have highlighted someone passes a mult-row return value but the first return value was a single row one (we are in `wrap_row` functon).

Is this clear?